src/decisionengine/framework/taskmanager/TaskManager.py

```diff
@@ -180,66 +178,59 @@ def _make_workers_for(configs, base_class, channel_name):
     return {key: Worker(key, e, base_class, channel_name) for key, e in configs.items()}
 
 
-class Workflow:
+class TaskManager(ComponentManager):
     """
-    Decision Channel.
-    Instantiates workers according to channel configuration
+    Task manager
     """
 
-    def __init__(self, channel_dict, channel_name):
+    def __init__(self, name, channel_dict, global_config, de_source_workers=None):
         """
+        :type name: :obj:`str`
+        :arg name: Name of channel corresponding to this task manager
+        :type generation_id: :obj:`int`
+        :arg generation_id: Task manager generation id provided by caller
         :type channel_dict: :obj:`dict`
         :arg channel_dict: channel configuration
+        :type global_config: :obj:`dict`
+        :arg global_config: global configuration
         """
+        super().__init__(channel_dict.get("channel_name", name))
 
-        delogger.debug("Creating channel sources")
-        self.source_workers = _make_workers_for(channel_dict["sources"], Source, channel_name)
+        self.id = str(uuid.uuid4()).upper()
+        self.dataspace = dataspace.DataSpace(global_config)
+        self.data_block_t0 = datablock.DataBlock(self.dataspace, name, self.id, 1)  # my current data block
+        self.logger = structlog.getLogger(CHANNELLOGGERNAME)
+        self.logger = self.logger.bind(module=__name__.split(".")[-1], channel=self.name)
+
+        self.broker_url = global_config.get("broker_url", "redis://localhost:6379/0")
```

Comment:
Did we want to log the `broker_url` in use here at `debug` or would that be too chatty?

Reply:
Logging it is a good idea.